tests/src/Integration/Condition/EntityIsNewTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Condition\EntityIsNewTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Condition;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Condition\EntityIsNew
+ * @group rules_conditions
+ */
+class EntityIsNewTest extends RulesIntegrationTestBase {
+
+  /**
+   * The condition to be tested.
+   *
+   * @var \Drupal\rules\Engine\RulesConditionInterface
+   */
+  protected $condition;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->condition = $this->conditionManager->createInstance('rules_entity_is_new');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary()
+   */
+  public function testSummary() {
+    $this->assertEquals('Entity is new', $this->condition->summary());
+  }
+
+  /**
+   * Tests evaluating the condition.
+   *
+   * @covers ::evaluate()
+   */
+  public function testConditionEvaluation() {
+    $entity = $this->getMock('Drupal\Core\Entity\EntityInterface');
+    $entity->expects($this->once())
+      ->method('isNew')
+      ->will($this->returnValue(TRUE));
+
+    // Add the test node to our context as the evaluated entity.
+    $this->condition->setContextValue('entity', $this->getMockTypedData($entity));
```

Comment:
That's bad practice (and needs to be removed from more tests).
setContextValue() is fine with $entity, nothing else required.